src/org/traccar/FilterHandler.java

```diff
@@ -150,16 +150,40 @@ private boolean filterLimit(Position position) {
 
     private boolean filter(Position p) {
 
+        String filterType = "";
+
         boolean result = filterInvalid(p) || filterZero(p) || filterDuplicate(p)
                 || filterFuture(p) || filterApproximate(p) || filterStatic(p)
                 || filterDistance(p);
 
+        if (filterInvalid(p)) {
```

Comment:
You are checking every filter twice. I think it's not a good idea. Filter check should be combined with logging.